packages/grafana-ui/src/components/Legend/Legend.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+
+import { StatID } from '../../utils/statsCalculator';
+import { DisplayValue } from '../../types/index';
+
+export interface LegendBasicOptions {
+  isVisible: boolean;
+  asTable: boolean;
+}
+
+export interface LegendRenderOptions {
+  placement: LegendPlacement;
+  hideEmpty?: boolean;
+  hideZero?: boolean;
+}
+
+export type LegendPlacement = 'under' | 'right' | 'over'; // Over used by piechart
+
+export interface LegendOptions extends LegendBasicOptions, LegendRenderOptions {
+  stats?: StatID[];
+  decimals?: number;
+}
+
+export interface StatDisplayValue extends DisplayValue {
+  statId: string;
+}
```

Comment:
rather than statId, what about either using `title` or making this:
```
export interface LegendDisplayValue extends DisplayValue {
  key: string;
}
```